orchagent/nhgorch.cpp

```diff
@@ -0,0 +1,1386 @@
+                                "Failed to sync temporary NHG %s with %s",
+                                index.c_str(),
+                                nhg_key.to_string().c_str());
+                        }
+                    }
+                    catch (const std::exception& e)
+                    {
+                        SWSS_LOG_WARN(
+                            "Got exception: %s while adding temp group %s",
+                            e.what(),
+                            nhg_key.to_string().c_str());
+                    }
+
+                    /*
+                     * We set the success to false so we keep trying to update
+                     * this group in order to sync the whole group.
+                     */
+                    success = false;
+                }
+                else
+                {
+                    auto nhg = std::make_unique<NextHopGroup>(nhg_key);
+                    success = nhg->sync();
+
+                    if (success)
+                    {
+                        SWSS_LOG_INFO("NHG successfully synced");
+                        m_syncdNextHopGroups.emplace(index,
+                                                    NhgEntry(std::move(nhg)));
+                    }
+                }
+            }
+            /* If the group exists, update it. */
+            else
+            {
+                SWSS_LOG_NOTICE("Update next hop group %s with %s",
+                                index.c_str(),
+                                nhg_str.c_str());
+
+                const auto& nhg_ptr = nhg_it->second.nhg;
+
+
+                /*
+                 * A NHG update should never change the SAI ID of the NHG if it
+                 * is still referenced by some other objects, as they would not
+                 * be notified about this change.  The only exception to this
+                 * rule is for the temporary NHGs, as the referencing objects
+                 * will keep querying the NhgOrch for any SAI ID updates.
+                 */
+                if (!nhg_ptr->isTemp() &&
+                    ((nhg_key.getSize() == 1) || (nhg_ptr->getSize() == 1)) &&
+                    (nhg_it->second.ref_count > 0))
+                {
+                    SWSS_LOG_WARN("Next hop group %s update would change SAI "
+                                  "ID while referenced, so not performed",
+                                  index.c_str());
+                    success = false;
+                }
+                /*
+                 * If the update would mandate promoting a temporary next hop
+                 * group to a multiple next hops group and we do not have the
+                 * resources yet, we have to skip it until we have enough
+                 * resources.
+                 */
+                else if (nhg_ptr->isTemp() &&
+                         (nhg_key.getSize() > 1) &&
+                         (NextHopGroup::getCount() >= m_maxNhgCount))
```

Comment:
count check to include next hop groups created by routeOrch

Reply:
Ditto from above - this already counts all next hop groups